Index: framework/common/widget/CommonScreens.xml
===================================================================
--- framework/common/widget/CommonScreens.xml	(revision 1294940)
+++ framework/common/widget/CommonScreens.xml	(working copy)
@@ -489,6 +489,7 @@
         <section>
             <actions>
                 <set field="layoutSettings.commonHeaderImageUrl" value="/images/ofbiz_logo.gif"/>
+                <set field="defaultFontFamily" value="DejaVuSans"/>
             </actions>
             <widgets>
                 <platform-specific>
Index: framework/webapp/config/fop.xconf
===================================================================
--- framework/webapp/config/fop.xconf	(revision 1294940)
+++ framework/webapp/config/fop.xconf	(working copy)
@@ -84,7 +84,10 @@
                           | 500 | 600 | 700 | 800 | 900
         (normal = 400, bold = 700)
         -->
-
+        <fo:inline font-family="Loma" font-weight="bold" font-style="normal">
+            Loma-normal-normal font
+        </fo:inline>
+        <font-triplet name="Loma" style="normal" weight="bold"/>
         <!--
         <font metrics-url="arial.xml" kerning="yes" embed-url="arial.ttf">
           <font-triplet name="Arial" style="normal" weight="normal"/>
@@ -95,6 +98,19 @@
           <font-triplet name="ArialMT" style="normal" weight="bold"/>
         </font>
         -->
+        <!--Loma -->
+        <font metrics-url="fonts/Loma.xml" kerning="yes" embed-url="fonts/Loma.ttf">
+          <font-triplet name="Loma" style="normal" weight="normal"/>
+        </font>
+        <font metrics-url="fonts/Loma-Bold.xml" kerning="yes" embed-url="fonts/Loma-Bold.ttf">
+          <font-triplet name="Loma" style="normal" weight="bold"/>
+        </font>
+        <font metrics-url="fonts/Loma-Oblique.xml" kerning="yes" embed-url="fonts/Loma-Oblique.ttf">
+          <font-triplet name="Loma" style="italic" weight="normal"/>
+        </font>
+        <font metrics-url="fonts/Loma-BoldOblique.xml" kerning="yes" embed-url="fonts/Loma-BoldOblique.ttf">
+          <font-triplet name="Loma" style="italic" weight="bold"/>
+        </font>
       </fonts>
 
       <!-- This option lets you specify additional options on an XML handler -->
